### DISCUSSION TITLE
No matching client found for package name 'capacitor.android.plugins'

### DISCUSSION THREAD

**Discussion by hnassr** [ionic-team/capacitor#1286] [Q&A]
When i try to run app on android i get this error `No matching client found for package name 'capacitor.android.plugins'`

the problem appear after installing this package `cordova-plugin-fcm-with-dependency-updated`

**Comment by zakton5** [comment]
@hnassr Did you find a solution?

**Comment by hnassr** [comment]
@zakton5 no, and i currently don't care  😅

**Comment by emi-arcioni** [comment]
Hey @zakton5 , did you find any solution?

**Comment by ga27** [comment]
Same problem, someone find a solution? @florianchevallier sadly your solution don't work for me

**Comment by ujeon** [comment]
Did you solve this problem? I'm in emergency

**Comment by ga27** [comment]
> Did you solve this problem? I'm in emergency

I've switched from FCM to PushNotifications plugin in Capacitor core. It works with some bugs, but it works.

**Comment by Radecom** [comment]
I get the same problem installing cordova-plugin-fcm-with-dependency-updated with capacitor, the funny thing is that only with capacitor, since with cordova it works very well.

Does anyone know the trick to install cordova-plugin-fcm-with-dependency-updated with capacitor?

**Comment by jcesarmobile** [comment]
see https://github.com/ionic-team/capacitor/issues/2483

**Comment by rathorerinki** [comment]
I've also same issue below still link please give me any suggestion 
https://user-images.githubusercontent.com/48315035/86244041-f9f83980-bbc4-11ea-98eb-efa95b5dfcb9.png

**Comment by nisith44** [comment]
i found solution ..
1st method
open the manifest file in "capacitor-cordova-android-plugins"  folder in android studio
replace "capacitor.android.plugins"  with your app package name..
now FCM work for me..
**update** first method not work for release builds  

2nd method
goto your firebase account --> goto your project
click "add app" 
make new app with this package name "capacitor.android.plugins"
download the "google-services.json" 
copy it to this path of your project 
"android/capacitor-cordova-android-plugins"
"android/app"
now it should work..

**Reply by PabloASua** [reply]
It Works! Thaks!

**Reply by sunilksamanta** [reply]
Worked. Thank you for leting me sleep finally.

**Reply by Funmbie** [reply]
Very helpful thanks

**Reply by albertinigr** [reply]
It works, thanks you are a hero!

**Reply by ShanahJr** [reply]
Worked for me too! You are awesome!

**Reply by RafaSilice** [reply]
It does not work. When generating the release apk it gives the following error:

Type BuildConfig is defined multiple times

Solved. So that not of that error add in the build.gradle:

     buildFeatures {
          buildConfig = false
      }

**Reply by didinj** [reply]
@RafaSilice which build.gradle?

**Reply by RafaSilice** [reply]
build.grandle (Module: android.app)

**Reply by didinj** [reply]
@RafaSilice it works, I put that config in android/app/build.config
`android {
  ...
  buildFeatures {
    buildConfig = false
  }
}`

Thanks!
